twoliter/embedded/rpm2kit

```diff
@@ -30,3 +35,122 @@ done
 
 createrepo_c "${KIT_DIR}"
 dnf --disablerepo '*' --repofrompath "kit,file:///${KIT_DIR}" repoquery --all
+
+WORK_DIR="$(mktemp -d)"
+TIMESTAMP="$(date +"%FT%T.%NZ")"
+FILENAME_PREFIX="${KIT:?}-v${VERSION_ID:?}-${BUILD_ID:?}-${ARCH:?}"
+
+# Translate ARCH into the proper docker arch
+case "${ARCH}" in
+  "x86_64")
+  DOCKER_ARCH="amd64"
+  ;;
+  "aarch64")
+  DOCKER_ARCH="arm64"
+  ;;
+esac
+
+# Create the content layer
+mkdir -p "${WORK_DIR}/blobs/sha256"
+DIFF_ARCHIVE="${WORK_DIR}/content-layer.tar"
+tar -cf "${DIFF_ARCHIVE}" -C "${KIT_DIR}" .
+DIFF_ID="$(digest_from_file "${DIFF_ARCHIVE}")"
+gzip "${DIFF_ARCHIVE}"
+CONTENT_ARCHIVE="${DIFF_ARCHIVE}.gz"
+CONTENT_DIGEST="$(digest_from_file "${CONTENT_ARCHIVE}")"
+mv "${WORK_DIR}/content-layer.tar.gz" "${WORK_DIR}/blobs/sha256/${CONTENT_DIGEST}"
+
+METADATA_JQ=$(cat <<EOF
+{
+  name: "$KIT",
+  version: "$VERSION_ID",
+  sdk: .[0].sdk,
+  kit: (
+    [ .[1] | values[] | {name: ., version: "$VERSION_ID", vendor: "$VENDOR"}]
+    + [ .[0].kit[] | {name: .name, version: .version, vendor: .vendor } ]
+ )
+}
+EOF
+)
+declare -a "LOCAL_KITS=(${LOCAL_KIT_DEPENDENCIES})"
+LOCAL_KIT_LIST=$(jq -c -n '$ARGS.positional // []' --args "${LOCAL_KITS[@]}")
+METADATA=$({ cat "/host/${EXTERNAL_KIT_METADATA}"; echo "${LOCAL_KIT_LIST}"; } | jq -c --sort-keys -s "${METADATA_JQ}")
```

Comment:
maybe something along these lines to make the transformations and inputs more readable:
```suggestion
METADATA_TEMPLATE="..."
LOCAL_KIT_INPUT="..."
EXTERNAL_KIT_INPUT="$(cat "/host/${EXTERNAL_KIT_METADATA}")"

KIT_INPUT="{ ${EXTERNAL_KIT_INPUT} ${LOCAL_KIT_INPUT} }"
KIT_METADATA="$(jq --compact-output --sort-keys --slurp "${METADATA_TEMPLATE}" <<<"${KIT_INPUT}")"
```